File: DeckOfCards-Java/src/test/java/com/aprearo/deckofcards/CardTests.java
import static org.junit.Assert.*;

import org.junit.Test;

import com.aprearo.deckofcards.*;

public class CardTests {
    @Test
    public void testSuitCount() {
        assertEquals(Card.Suit.values().length, 4);
    }

    @Test
    public void testRankCount() {
        assertEquals(Card.Rank.values().length, 13);
    }

    @Test
    public void testEquals() {
        Card card1 = new Card(Card.Suit.DIAMONDS, Card.Rank.QUEEN);
        Card card2 = new Card(Card.Suit.DIAMONDS, Card.Rank.QUEEN);
        assertEquals(card1, card2);
        Card card3 = new Card(Card.Suit.DIAMONDS, Card.Rank.ACE);
        assertNotEquals(card1, card3);
    }

    @Test
    public void testGetSuit() {
        Card.Suit[] suits = Card.Suit.values();
        int count = suits.length;
        for (int i=0; i < count; ++i) {
            Card card = new Card(suits[i], Card.Rank.ACE);
            assertEquals(card.getSuit(), suits[i]);
            assertEquals(card.getRank(), Card.Rank.ACE);
        }
    }

    @Test
    public void testGetRank() {
        Card.Rank[] ranks = Card.Rank.values();
        int count = ranks.length;
        for (int i=0; i < count; ++i) {
            Card card = new Card(Card.Suit.DIAMONDS, ranks[i]);
            assertEquals(card.getSuit(), Card.Suit.DIAMONDS);
            assertEquals(card.getRank(), ranks[i]);
        }
    }
}

File: DeckOfCards-Kotlin/src/test/kotlin/com/aprearo/deckofcards/CardTests.kt
package com.aprearo.deckofcards

import org.junit.Test
import kotlin.test.*

class CardTests {
    @Test
    fun testSuitCount() {
        assertEquals(Suit.values().size, 4)
    }

    @Test
    fun testRankCount() {
        assertEquals(Rank.values().size, 13)
    }

    @Test
    fun testEquals() {
        val card1 = Card(Suit.DIAMONDS, Rank.QUEEN)
        val card2 = Card(Suit.DIAMONDS, Rank.QUEEN)
        assertEquals(card1, card2)
        val card3 = Card(Suit.DIAMONDS, Rank.ACE)
        assertNotEquals(card1, card3)
    }

    @Test
    fun testGetSuit() {
        val suits = enumValues<Suit>()
        val count = suits.size
        for (i in 0 until count) {
            val card = Card(suits[i], Rank.ACE)
            assertEquals(card.suit, suits[i])
            assertEquals(card.rank, Rank.ACE)
        }
    }

    @Test
    fun testGetRank() {
        val ranks = enumValues<Rank>()
        val count = ranks.size
        for (i in 0 until count) {
            val card = Card(Suit.DIAMONDS, ranks[i])
            assertEquals(card.suit, Suit.DIAMONDS)
            assertEquals(card.rank, ranks[i])
        }
    }
}
